### ISSUE TITLE
Authentication failed cloning public repo containing submodules

### ISSUE THREAD

**Issue by erikmartinessanches** [desktop/desktop#2344]
### Description

When cloning [my repo](https://github.com/erikmartinessanches/javadev) using HTTPS containing two submodules, I receive the following:
![githuberror](https://user-images.githubusercontent.com/4912935/28654672-7121356e-7296-11e7-9d21-af29f14a95ae.png)

- This error does not appear when using GitHub Desktop to clone the submodules individually. 

- This error does not appear when using terminal to clone my repo `git clone -recursive https://github.com/erikmartinessanches/javadev.git` but git prompts for password twice (for each submodule, how to enable cloning without password?)

### Version
git version 2.11.0 (Apple Git-81)

**GitHub Desktop version:** 0.7.0

**OS version:** Mac OS version 10.12.5 (16F73)

### Steps to Reproduce

1. Select Open in Desktop (HTTPS option) on [my repo](https://github.com/erikmartinessanches/javadev)
2. Optionally adjust directory details
3. Error appears

I've tried signing out of GitHub and GitHub Desktop and signing in again, does not resolve.

**Expected behavior:** Clone repo with or without submodules.

**Actual behavior:** Only the outer repo seems to get cloned, then the error above appears, the submodule repos are not cloned.

**Reproduces how often:** 100%

### Logs
[2017-07-27.desktop.production.log.zip](https://github.com/desktop/desktop/files/1178853/2017-07-27.desktop.production.log.zip)

**Comment by shiftkey** [comment]
@erikmartinessanches thanks for the detailed report (and public repository that demonstrates the issue):

> git prompts for password twice (for each submodule, how to enable cloning without password?)

I think this is due to the [submodules](https://github.com/erikmartinessanches/javadev/blob/master/.gitmodules) using SSH remotes (there's a slightly different flow here, which I can write more about after :coffee:).

As these submodules are public, they can be changed over to HTTPS URLs which will not trigger the authentication steps. Just remember to synchronize the submodules on your machine after the change:

```
git submodule sync
git submodule update --init --recursive --remote
```

**Comment by shiftkey** [comment]
Reproducible on macOS:

```
Submodule 'patterns' (git@github.com:erikmartinessanches/patterns.git) registered for path 'patterns'
Cloning into '/Users/shiftkey/Desktop/new/javadev/algorithms'...
ssh_askpass: exec(/usr/X11R6/bin/ssh-askpass): No such file or directory
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

It looks like our setup for prompting for credentials falls through to some random defaults when it encounters an SSH remote. Not sure if we need to opt in `SSH_ASKPASS` as per [these docs](https://git-scm.com/docs/gitcredentials#_requesting_credentials) or something else.

**Comment by shiftkey** [comment]
@erikmartinessanches apologies for the delay in replying. As a workaround you could [change these entries](https://github.com/erikmartinessanches/javadev/blob/master/.gitmodules) to be HTTPS URLs, which will bypass the need to invoke SSH for cloning the submodules which seems to have issues unless you have configured SSH credentials.

**Comment by ievgennaida** [comment]
I have the same issue with one sub-module. Terminal works fine.

**Comment by varunsinghal65** [comment]
Hello @erikmartinessanches,
I faced the exact same issue, for me even terminal was not working.
This was after my repo was configured to target HTTPS url and not SSH.

As i workaround i deleted the stored credentials (windows) : 
Credentials Manager -> Manage windows credentials -> delete all stored credentials for GIT

After this it worked from terminal as well as the UI (Git ext)

**Comment by fluffy** [comment]
Having exactly this issue. Hard to believe this has been open since 2017.

**Comment by sergiou87** [comment]
This seems related to the lack of proper support for SSH authentication in GitHub Desktop, but that should've been solved with #12756 and #14677 

I just tested it and works as expected with private SSH and HTTPS submodules.

**Comment by JosephTLyons** [comment]
This is happening to me right now.
